web/views/exposure.py

```diff
@@ -41,6 +43,8 @@ def form_valid(self, form):
         self.object.save()
         messages.add_message(self.request, messages.SUCCESS, 'exposure endpoint created')
         self.dataset.generate_elu_accession()
+        # publish subentities
+        self.dataset.publish()
```

Comment:
shouldn't we just put the call to self.dataset.generate_elu_accession() directly in dataset.publish to avoid having to call two functions and make sure the generate_elu_accession is always called with publish?

Reply:
i felt then we need to rename the method because it is somehow different logic

Reply:
I meant keeping the generate_elu_accession method, just calling it when we call publish. As we should never publish without assigning an elu accession id, for me, it makes sense, no?

Reply:
yes, will make it as before then

Reply:
done